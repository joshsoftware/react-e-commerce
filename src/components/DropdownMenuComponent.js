import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import DropdownItemComponent from './DropdownItemComponent';
import { DropdownMenu } from 'reactstrap';
import DropdownItemWrapper from './DropdownItemWrapper';
import { logoutRequest } from '../actions/formActions';

const DropdownMenuComponent = () => {
  const dispatch = useDispatch();
  const { userDetails } = useSelector((state) => state.loginReducer);

  return (
    <DropdownMenu right>
      <DropdownItemComponent option="Profile" />
      <DropdownItemWrapper
        option="Logout"
        onClick={() => {
          logoutRequest(userDetails.token, dispatch);
        }}
      />
    </DropdownMenu>
  );
};

export default DropdownMenuComponent;
