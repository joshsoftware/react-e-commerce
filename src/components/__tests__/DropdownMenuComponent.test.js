import DropdownMenuComponent from '../DropdownMenuComponent';
import { render } from '@testing-library/react';
import React from 'react';

describe('DropdownMenu Component', () => {
  it('Must return Dropdown Menu', () => {
    const { asFragment } = render(<DropdownMenuComponent />);

    expect(asFragment()).toMatchSnapshot();
  });
});
