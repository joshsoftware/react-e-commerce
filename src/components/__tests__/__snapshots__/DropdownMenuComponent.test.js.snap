// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownMenu Component Must return Dropdown Menu 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    class="dropdown-menu dropdown-menu-right"
    role="menu"
    tabindex="-1"
  >
    <button
      class="dropdown-item"
      role="menuitem"
      tabindex="0"
      type="button"
    >
      Profile
    </button>
    <div
      class="dropdown-divider"
      tabindex="-1"
    />
    <button
      class="dropdown-item"
      role="menuitem"
      tabindex="0"
      type="button"
    >
      Logout
    </button>
    <div
      class="dropdown-divider"
      tabindex="-1"
    />
  </div>
</DocumentFragment>
`;
